### DISCUSSION TITLE
How to avoid history entry for new iframe src

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4334] [1. Help]
I have an iframe which changes its source dynamically. Now, everytime a new source is loaded, the browser history makes a new entry. How can I avoid that? Is there maybe a way to destroy and recreate the iframe with the new src instead of changing it?

**Comment by ekwoka** [comment]
You could use x-if.

or even an x-for with a list of just the one src as the id. Then it would destroy and create a new one.

**Comment by simonferndriger** [comment]
It's interesting, I already had the `<template x-if>`, and also tried the `<template x-for>`, but since the value just got replaced immediately, it kept the iframe. Now I just use a `setTimeout()` (after setting it empty first) without actual delay, and it did the trick. You brought me on the right track :)

**Reply by SimoTod** [reply]
I think you can also have `x-effect="$el.contentDocument.location.replace(url)"` on the iframe with url being your reactive variable.

**Reply by ekwoka** [reply]
`<template x-for>` requires that you key on something other than the index, like the actual value. then it would destroy the element and make a new one.

**Reply by simonferndriger** [reply]
@ekwoka  Ah, OK, thanks

@SimoTod That also sounds interesting - I really like this approach since you could avoid recreating the iframe (which is time consuming and shows up in a little bit of glitch when navigating back and forth). Unfortunately, it didn't work, sorry...
